pkg/kstatus/status/status_compute_test.go

```diff
@@ -1269,6 +1269,85 @@ func TestJobStatus(t *testing.T) {
 	}
 }
 
+var hpaNoStatus = `
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: test
+  namespace: qual
+  generation: 1
+`
+
+var hpaCurrentStatus = `
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: test
+  namespace: qual
+  generation: 1
+status:
+  conditions:
+  - type: AbleToScale
+    status: "True"
+  - type: ScalingActive
+    status: "True"
+`
+
+var hpaScalingLimitedStatus = `
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: test
+  namespace: qual
+  generation: 1
+status:
+  conditions:
+  - type: AbleToScale
+    status: "True"
+  - type: ScalingActive
+    status: "True"
+  - type: ScalingLimited
+    status: "True"
+`
+
+func TestHPAStatus(t *testing.T) {
+	testCases := map[string]testSpec{
+		"hpaNoStatus": {
+			spec:               hpaNoStatus,
+			expectedStatus:     UnknownStatus,
```

Comment:
Usually when an object doesn't have status, it's shown as Current, with the assumption that there might not be any status. The way to work around that is to default `status.observedGeneration=0` in the CRD. Is that not true for the HPA CRD?

I wonder if we should get that changed upstream.